3MI0800303/practicum-1.cpp

```diff
@@ -0,0 +1,148 @@
+#include <iostream>
+#pragma warning(disable : 4996)
+
+struct OrderedTwo {
+    int firstNumber;
+    int secondNumber;
+    bool isBigger(OrderedTwo& ordered) const;
+};
+
+inline bool OrderedTwo::isBigger(OrderedTwo& cmp) const {
+    if (firstNumber == cmp.firstNumber) return secondNumber > cmp.secondNumber;
+    if (firstNumber < cmp.firstNumber) return false;
+    return true;
+}
+
+int getBiggest(OrderedTwo ords[100]) {
+    for (size_t i = 0; i < 100; i++) {
+        for (size_t j = i + 1; j < 100; j++) {
+            if (ords[i].isBigger(ords[j])) std::swap(ords[i], ords[j]);
+        }
+    }
+
+    return 0;
+}
+
+enum class Course {
+    first,
+    second,
+    third,
+    fourth
+};
+
+struct Student {
+    char name[16];
+    int fn_number;
+    Course course;
+    double avg_grade;
+    void printStudents() const;
+};
+
+void Student::printStudents() const {
+    char courseChar[7];
+    switch (course) {
+    case Course::first: strcpy(courseChar, "first"); break;
+    case Course::second: strcpy(courseChar, "second"); break;
+    case Course::third: strcpy(courseChar, "third"); break;
+    case Course::fourth: strcpy(courseChar, "fourth"); break;
+    }
+
+    std::cout << "Name: " << name << " Fn Number: " << fn_number << " course: " << courseChar << " grade: " << avg_grade << std::endl;
+    //std::cout << "Name:" << x << " Fn Number: %"
+}
+
+bool initStudent(Student& toInit, char name[16], int facNumber, Course currentCourse, double grade) {
+    if (facNumber <= 99999 || facNumber > 999999) return false;
```

Comment:
This hasn't really been covered yet (exceptions & error handling) but having a boolean return value to signal the success / failure of an operation (in this case - the initialization of an object) is not a good idea. Boolean values don't really mean much (if `false` nobody will be able to tell the tale of what exactly went wrong) and can easily be overlooked (e.g. if I don't do an explicit check about the given return value - I'll never now that an error occurred). Keep in mind that [some other mechanisms](https://www.tutorialspoint.com/cplusplus/cpp_exceptions_handling.htm) exist for doing just that :))